[26기 김여름] React: Refectoring

sunmerrr opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [x] 리뷰 반영
- [x] 리팩토링
- [x] 버그 수정
- [x] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- Refactoring
- id, pw 입력 이벤트 수정
- 이메일 검증 시 '@'가 아닌 정규표현식 사용
- 댓글 data에 따른 isliked 활성
- 2개로 나눠져있던 mock data를 하나로 합침
- mock data에 넣어준 값을 통해서 like, id, 본문, 댓글 등 반영
- props 정리
- 회원가입, 로그인 기능 구현
- 변수 이름 수정, 변수 단축, scss 수정

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. Refactoring
 - 콘솔로그 제거, 필요없는 주석 제거 중
2. id, pw 입력 이벤트 수정
 - input입력 이벤트 두 개로 나눠놨던 것을 하나로 합침
3. 이메일 검증 시 '@'가 아닌 정규표현식 사용
 - 정규 표현식을 사용하여 id 입력값이 이메일 형식이 맞는지 검증하는 코드를 구현함
 - 하지만 이메일 형식이 맞지 않는다고해서 경고창이 뜨지는 않음
4. 댓글 data에 따른 isliked 활성
 - data에 따른 컴포넌트 상태 변경
5. 2개로 나눠져있던 mock data를 하나로 합침
 - 기존에 두개로 나눠서 넣어줬던 데이터를 하나로 합쳐서 데이터에 있는 좋아요, 작성자, 본문, 댓글, 댓글 내용, 댓글 좋아요 등 실현
6. props 정리
 - 기존에 `props`를 `this.props.propsName`으로 전달해줬던 값을 구조분해 할당하여 코드가 읽히기 쉽게 정리함
7. 회원가입 로그인 구현
 - 백엔드의 API를 통해서 회원가입을 하고, 가입한 정보로 로그인 할 수 있도록 구현하였습니다.

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 정규표현식을 사용하여 건증하는 법을 배웠습니다.
- input 이벤트를 input값마다 달아주는 것이 아니라 하나의 이벤트로 핸들링 하는 법을 배웠습니다.
- data에 따른 변화를 inline style이 아닌 css 값으로 변경해주었습니다.
- 구조분해 할당을 활용하여 props를 깔끔하게 볼 수 있도록 하였습니다.
- 회원가입 로그인 시 백엔드와 통신 방법, 통신 상태, fetch 다루는 방법, token 저장 방법 등을 배웠습니다.

<br />

## :: 기타 질문 및 특이 사항
* 특이사항
- 유효성 검사 후 메세지 문제 해결 예정

sunmerrr committed:
Mission 1, 2: 로그인 페이지 추가 기능 구현 완료

sunmerrr committed:
Fix: conflict resolved
Mission 3,4: 댓글 기능 구현, 댓글 컴포넌트화 + props로 전달

Merge branch 'master' into feature/summer

sunmerrr committed:
Mission: Mock Data 활용

sunmerrr committed:
Refactoring: login page 진행중

sunmerrr committed:
refactoring: 로그인화면 및 메인 화면 리펙토링중

brparkk commented on src/pages/SummerKIm/Login/Login.js:
```diff
@@ -1,13 +1,72 @@
 
-export class Nav extends Component {
   render() {
+    let regExp =
```
정규표현식으로 사용하는 게 더 좋을거같네요! 참고해서 저도 수정하겠습니다~!!🙂👍👍👍👍👍

sunmerrr replied:
네! 저도 처음 사용해본거라서 많이 미숙하네요ㅎㅎ 제가 쓴 것 말고 다른 방법이 있다면 공유드릴게욤🙇‍♀️

JeonChangMin15 commented on src/pages/SummerKIm/Login/Login.js:
```diff
@@ -1,13 +1,72 @@
+              style={{
+                opacity: idVar && pwVar ? '1' : '0.4',
+              }}
+              disabled={!idVar && !pwVar}
```
불리언으로 간단하게 구현한게 좋네요~~

sunmerrr replied:
감사합니다 😆🙇‍♀️

okcleff commented on src/pages/SummerKIm/Login/Login.js:
```diff
@@ -1,13 +1,72 @@
+              style={{
+                opacity: idVar && pwVar ? '1' : '0.4',
+              }}
+              disabled={!idVar && !pwVar}
```
변수와 논리연산자로 `disabled`의 속성값을 부여하신게 인상 깊어요! 멋지십니다.

sunmerrr replied:
감사합니다😁🙇‍♀️

JeonChangMin15 commented on src/pages/SummerKIm/Main/Feed/Feed.js:
```diff
@@ -0,0 +1,117 @@
+      });
+      // 새로 작성되는 댓글은 commentList 에 들어가긴 하나, data로 받아오는 형식과 다르기때문에 comment가 자식 component에 전달되지는 못한다.
+    }
+    this.commentValue = '';
```
삭제해도 괜찮을거 같아요

sunmerrr replied:
넵 반영하도록 하겠습니다! 💪

JeonChangMin15 reviewed (commented):
고생하셨습니다 여름님!

brparkk reviewed (commented):
여름님 수고하셨습니다~~😺🤟🤟 코멘트할게 없어여 역시 갓벽하시네유

okcleff commented on src/pages/SummerKIm/Main/MainRight/MainRight.js:
```diff
@@ -0,0 +1,116 @@
+        <div className="suggestion">
+          <span className="topofBox">스토리</span>
+          <button className="storyAll">모두 보기</button>
+          <ur className="stories">
```
`<ur>` 태그가 오타인 것 같아요! 확인 부탁 드립니다 ^^

sunmerrr replied:
오타 체크 감사드립니다!! 반영하도록 하겠습니다💪

okcleff reviewed (commented):
여름님 오늘도 한 수 배우고 갑니다! 화이팅 하세요!

sunmerrr committed:
refectoring: peer review 반영 및 코드 수정
- console.log 제거
- 오타 수정
- props 정리
- comment mock data 를 feed mock data에 합친 후 data활용

sunmerrr committed:
Add: Signup page
refactoring: inline style 수정중

sunmerrr committed:
refectoring: 회원가입 페이지 수정

sunmerrr committed:
refectoring: 로그인 페이지

sunmerrr committed:
refectoring: feed
